servicetalk-serializer-utils/src/main/java/io/servicetalk/serializer/utils/AbstractStringSerializer.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.serializer.utils;
+
+import io.servicetalk.buffer.api.Buffer;
+import io.servicetalk.buffer.api.BufferAllocator;
+import io.servicetalk.serializer.api.SerializerDeserializer;
+
+import java.nio.charset.Charset;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.servicetalk.utils.internal.CharsetUtils.standardCharsets;
+
+abstract class AbstractStringSerializer implements SerializerDeserializer<String> {
```

Comment:
Serializer/Deserializer in name? same for subclass

Reply:
A shorthand in the form of SerDe is frequently used in the community.

Reply:
The strategy I used was to be explicit with the interface names (`SerializerDeserializer`, `BufferEncoderDecoder`) but just use the "serializer" variant for concrete method/type names that fill both roles for brevity (`StringSerializer`, `NettyBufferEncoders#bufferEncoder(..)`, `HttpSerializers#textSerializerUtf8()`). In practice I'm not sure lengthening the names of concrete types/methods, or using abbreviations (`SerDe`), will improve understandability/readability.